.github/workflows/ci.yml

```diff
@@ -99,11 +114,21 @@ jobs:
           echo "=== Post-deployment validation ==="
           ./scripts/test.sh check-deployment
 
+      - name: Install Python dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install pytest requests psycopg2-binary
```

Comment:
I think these can be installed in the pip install further up, rather than having two install steps